logdna/tests/test.js

```diff
@@ -0,0 +1,72 @@
+const index = require('../index.js');
+
+let asyncRetry = require('async').retry;
+let request = require('request');
+jest.mock('async');
+jest.mock('request');
+
+const testCallback = jest.fn();
+const errorObject = {code: 'testError'};
+const mockRes = {responseCode: 200};
+const mockBody = {body: 'mockBody'};
+const awsEvent = {
+    awslogs: {
+        /* eslint-disable max-len */
+        data: 'H4sIAAAAAAAAEzWQQW+DMAyF/wrKmaEkJCbhhjbWCzuBtMNUVSmkNBIQRMKqqep/X6Cb5Ivfs58++45G7ZzqdfMza5Sjt6IpTh9lXReHEsXI3ia9BJnQlHHIhMSEBnmw/WGx6xwcp8Z50M9uN2q/aDUGx2vn/5oYufXs2sXM3tjp3QxeLw7lX6hS47lTz6lTO9i1uynfXkOMe5lsp9Fxzyy/9eS3hTsyXYhOGVCaEsBSgsyEYBkGzrDMAIMQlAq+gQIQSjFhBFgqJOUMAog34WAfoFFOOM8kA0Y5SSH+f0SIb67GRaHq/baosn1UmUlHF7tErxvk5wa56b2Z+iRJ0OP4+AWj9ITzSgEAAA=='
+        /* eslint-enable max-len */
+    }
+};
+process.env.LOGDNA_KEY = 'keykey';
+describe('making a request', function() {
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+    it('calls the cb with the error if the request sends an error', function() {
+        index.handler(awsEvent, {}, testCallback);
```

Comment:
In that case, we won't need to test `index.handler` but `cloudwatch` class methods

Reply:
@smusali If I understand you correctly, you want to have tests for cloudWatch module separately because index.handler will be shared in the future for multiple modules. I agree with that. But now the project does not have anything besides cloudWatch and this kind of test works better now. I suggest changing it to be modular when it will explicitly need that.

Reply:
@vilyapilya, that makes sense now! Thanks for the explanation!